engine/frontend/src/component/Enclaves.js

```diff
@@ -12,43 +12,91 @@ import {
 
-const DeleteAlertDialog = ({isOpen, cancelRef, onClose, enclaveName, setEnclaveName, handleDeleteClick}) => {
+const DeleteAlertDialog = ({isOpen, cancelRef, onClose, enclaveToDelete, setEnclave, handleDeleteClick}) => {
     const [deleting, setDeleting] = useState(false);
-
+    const [value, setValue] = useState("")
+    const [error, setError] = useState(false)
+    
+    const enclaveName = enclaveToDelete.name;
     const handleClose = (action) => {
-        const maybeDeleteRequest = async (action) => {
+
+        const clickCancel = () => {
+            setError(false)
+            setEnclave({})
+            onClose()
+        }
+
+        const clickDelete = async () => {
+            setDeleting(true)
+            await handleDeleteClick(enclaveName)
+            setDeleting(false)
+            setError(false)
+            clickCancel()
+        }
+
+        const maybeDeleteRequest = async () => {
             if (action === "delete") {
-                setDeleting(true)
-                await handleDeleteClick(enclaveName)
-                setDeleting(false)
+                if (value === enclaveName) {
+                    await clickDelete()
+                } else {
+                    setError(true)
+                    setValue("")
+                }
+            } else {
+                clickCancel()
             }
-            setEnclaveName("")
-            onClose()
         }
-        maybeDeleteRequest(action)
+
+        maybeDeleteRequest()
+    }
+
+    const handleInputChange = (val) => {
+        if (error) {
+            setError(false)
+        }
+        setValue(val)
     }
 
     return (
         <AlertDialog
             isOpen={isOpen}
             leastDestructiveRef={cancelRef}
-            onClose={onClose}
+            onClose={ () => {
+                setError(false)
+                setValue("");
+                onClose();
+            }}
             isCentered
       >
         <AlertDialogOverlay>
-          <AlertDialogContent>
-            <AlertDialogHeader fontSize='lg'>
+          <AlertDialogContent backgroundColor={"white"}>
+            <AlertDialogHeader fontSize='lg' color={"black"}>
               Delete Enclave: <Text fontSize='lg' fontWeight='bold' as='i'> {enclaveName} </Text>
             </AlertDialogHeader>
 
             <AlertDialogBody>
-              Are you sure? You can't undo this action afterwards.
+                {
+                    enclaveToDelete.mode ? 
+                        <FormControl>
+                            <Input onChange={(e) => handleInputChange(e.target.value)} isInvalid={error} borderColor={"black"} color={"black"}/>
+                            <FormHelperText color={error ? "red.600" : "black"} fontSize={"sm"}> 
+                                {error ? 
+                                    "Please make sure that the input matches the enclave name" : 
+                                    "Please enter the enclave name to delete the enclave."
```

Comment:
"Enter the enclave name to delete the enclave"

skip also the period at the end